applovin_max/android/src/main/java/com/applovin/applovin_max/AppLovinMAX.java

```diff
@@ -1326,6 +1336,89 @@ else if ( adViewPosition.contains( "right" ) )
+        adInfo.put( "networkName", ad.getNetworkName() );
+        adInfo.put( "placement", !TextUtils.isEmpty( ad.getPlacement() ) ? ad.getPlacement() : "" );
+        adInfo.put( "revenue", ad.getRevenue() );
+        adInfo.put( "dspName", !TextUtils.isEmpty( ad.getDspName() ) ? ad.getDspName() : "" );
+        adInfo.put( "waterfall", createAdWaterfallInfo( ad.getWaterfall() ) );
+
+        return adInfo;
+    }
+
+    // AD WATERFALL INFO
+
+    private Map<String, Object> createAdWaterfallInfo(final MaxAdWaterfallInfo waterfallInfo)
+    {
+        Map<String, Object> waterfallInfoObject = new HashMap<>();
+        if ( waterfallInfo == null ) return waterfallInfoObject;
+
+        waterfallInfoObject.put( "name", waterfallInfo.getName() );
+        waterfallInfoObject.put( "testName", waterfallInfo.getTestName() );
+
+        List<Object> networkResponsesArray = new ArrayList<>();
+        for ( MaxNetworkResponseInfo response : waterfallInfo.getNetworkResponses() )
+        {
+            networkResponsesArray.add( createNetworkResponseInfo( response ) );
+        }
+        waterfallInfoObject.put( "networkResponses", networkResponsesArray );
+
+        waterfallInfoObject.put( "latencyMillis", waterfallInfo.getLatencyMillis() );
+
+        return waterfallInfoObject;
+    }
+
+    private Map<String, Object> createNetworkResponseInfo(final MaxNetworkResponseInfo response)
+    {
+        Map<String, Object> networkResponseObject = new HashMap<>();
+        networkResponseObject.put( "adLoadState", response.getAdLoadState().ordinal() );
+
+        MaxMediatedNetworkInfo mediatedNetworkInfo = response.getMediatedNetwork();
+        if ( mediatedNetworkInfo != null )
+        {
+            Map<String, String> networkInfoObject = new HashMap<>( 4 );
+            networkInfoObject.put( "name", mediatedNetworkInfo.getName() );
+            networkInfoObject.put( "adapterClassName", mediatedNetworkInfo.getAdapterClassName() );
+            networkInfoObject.put( "adapterVersion", mediatedNetworkInfo.getAdapterVersion() );
+            networkInfoObject.put( "sdkVersion", mediatedNetworkInfo.getSdkVersion() );
+
+            networkResponseObject.put( "mediatedNetwork", networkInfoObject );
+        }
+
+        Bundle credentialBundle = response.getCredentials();
+        Map<String, String> credentials = new HashMap<>();
+        for ( String key : credentialBundle.keySet() )
+        {
+            Object obj = credentialBundle.get( key );
+            if ( obj instanceof String )
+            {
+                credentials.put( key, (String) obj );
+            }
+        }
+        networkResponseObject.put( "credentials", credentials );
+
+        MaxError error = response.getError();
+        if ( error != null )
+        {
+            Map<String, Object> errorObject = new HashMap<>( 3 );
+            errorObject.put( "message", error.getMessage() );
+            errorObject.put( "adLoadFailureInfo", error.getAdLoadFailureInfo() );
```

Comment:
We actually deprecated this API and advise pubs to just use the waterfall API since it's more OOP + more info, so you can omit this (and we'll remove it from RN and other plugins in the future as well).

Reply:
Removed.